Feature/t3d

nickgal opened the pull request.

nickgal committed:
.NET Upgrade Assistant to dotnet 6

nickgal committed:
Merge branch 'master' into feature/dotnet6

nickgal committed:
Fix halas female full path

nickgal committed:
Init T3d archives & older fragments

nickgal committed:
Apply material fixer to all instances
Multiple fragments can have a reference to the same name.
For example qrg_obj has two materials that reference BARRELTOP_MDF

nickgal committed:
Support ungrouped face materials

nickgal committed:
Support 0x2E animated vertices

nickgal committed:
Fix beta beetle bone names

nickgal committed:
Merge branch 'development/0.1.7' of github.com:LanternEQ/LanternExtractor into feature/t3d

nickgal committed:
Merge branch 'development/0.1.7' of github.com:LanternEQ/LanternExtractor into feature/t3d

danwilkins committed:
Fix extraction from shortname

danwilkins commented on LanternExtractor/EQ/Archive/ArchiveType.cs:
```diff
@@ -0,0 +1,9 @@
+{
+    public enum ArchiveType
+    {
+        Unknown,
```
How often do we actually run into an `Unknown` type archive? Also, can you assign numerical values to these.

nickgal replied:
Realistically I think unknown will really only get hit when non archive files get renamed to one of the archive extensions or when the files are damaged to the point where the the first 4 (or second 4) bytes are incorrect.

danwilkins commented on LanternExtractor/EQ/Wld/DataTypes/RenderInfo.cs:
```diff
@@ -0,0 +1,80 @@
+
+            var ba = new BitAnalyzer(renderInfo.Flags);
+
+            var hasPen = ba.IsBitSet(0);
```
Do we know yet what this does? See it in the later WLD files, too.

nickgal replied:
There are a couple functions suggesting something related to palette & transparency but I haven't figured out how it all comes together yet. `t3dSetRenderContextTransparentPen` `t3dPaletteClosestPenToColor`

danwilkins committed:
Update local variable name to make more sense

danwilkins commented on LanternExtractor/EQ/Wld/Helpers/CharacterFixer.cs:
```diff
@@ -389,6 +390,29 @@ private void FixHalasFemale()
+        {
+            var skeleton = _wld.GetFragmentByName<SkeletonHierarchy>("BET_HS_DEF");
+
+            if (skeleton == null)
```
This is a good pattern we can apply to the others fixes which iterate through and continue on mismatched names.

danwilkins commented:
Excellent work on this!

danwilkins committed:
Apply magenta filtering to all non indexed masked images

nickgal committed:
Fix unwanted loading of chequip
Regression in https://github.com/LanternEQ/LanternExtractor/commit/9e67b3ca09589326f93967ea0a3baceeb1c91ff8

danwilkins committed:
Fix chequip being used when exporting characters

nickgal committed:
PR feedback

nickgal committed:
Add pri/sec mesh distinction to exports
